src/parse.y

```diff
@@ -765,6 +765,10 @@ expr:
 				reporter->Error("index slice assignment may not be used"
 				                      " in arbitrary expression contexts, only"
 				                      " as a statement");
+			// Type types cannot be reassigned because that could require parse-time
```

Comment:
Can you do this instead by marking the identifier as `const`? That might be a bit cleaner/more comprehensive (but I haven't thought much about this so could have something wrong)